options/cf_options.cc

```diff
@@ -94,9 +94,29 @@ static Status ParseCompressionOptions(const std::string& value,
         ParseInt(value.substr(start, value.size() - start));
     end = value.find(':', start);
   }
-  // parallel_threads is not serialized with this format.
-  // We plan to upgrade the format to a JSON-like format.
-  compression_opts.parallel_threads = CompressionOptions().parallel_threads;
+
+  // parallel_threads is optional for backwards compatibility
+  if (end != std::string::npos) {
+    start = end + 1;
+    if (start >= value.size()) {
+      return Status::InvalidArgument(
+          "unable to parse the specified CF option " + name);
+    }
+    // Since parallel_threads comes before enabled but was added optionally
+    // later, we need to check if this is the final token (meaning it is the
+    // enabled bit), or if there is another token (meaning this one is
+    // parallel_threads)
+    end = value.find(':', start);
+    if (end != std::string::npos) {
+      compression_opts.parallel_threads =
+          ParseInt(value.substr(start, value.size() - start));
+    } else {
+      // parallel_threads is not serialized with this format, but enabled is
+      compression_opts.parallel_threads = CompressionOptions().parallel_threads;
+      compression_opts.enabled =
+          ParseBoolean("", value.substr(start, value.size() - start));
+    }
+  }
 
   // enabled is optional for backwards compatibility
```

Comment:
Maybe we should document somewhere that `enabled` is actually required if the user tries to provide `parallel_threads`. It seems a bit difficult to figure out right now.

Reply:
> Maybe we should document somewhere that `enabled` is actually required if the user tries to provide `parallel_threads`. It seems a bit difficult to figure out right now.

You mean, explain it in the HISTORY.md?